### PR TITLE
Minor Release v1.1.1

### PR DESCRIPTION
**Pull request by tcojean**
## TODO
+ [x] fix sonarqube step problem
+ [x] run full pipeline and generate documentation
+ [x] ensure everything compiles properly on Windows
+ [x] proofread documentation and changelog

## CHANGELOG

This version of Ginkgo provides a few fixes in Ginkgo's core routines. The
supported systems and requirements are unchanged from version 1.1.0.

### Fixes
+ Improve Ginkgo's installation and fix the `test_install` step ([#406](https://github.com/ginkgo-project/ginkgo/pull/406)),
+ Fix some documentation issues ([#406](https://github.com/ginkgo-project/ginkgo/pull/406)),
+ Fix multiple code issues reported by sonarqube ([#406](https://github.com/ginkgo-project/ginkgo/pull/406)),
+ Update the git-cmake-format repository ([#399](https://github.com/ginkgo-project/ginkgo/pull/399)),
+ Improve the global update header script ([#390](https://github.com/ginkgo-project/ginkgo/pull/390)),
+ Fix broken bounds checks ([#388](https://github.com/ginkgo-project/ginkgo/pull/388)),
+ Fix CSR strategies and improve performance ([#379](https://github.com/ginkgo-project/ginkgo/pull/379)),
+ Fix a small typo in the stencil examples ([#381](https://github.com/ginkgo-project/ginkgo/pull/381)),
+ Fix ELL error on small matrices ([#375](https://github.com/ginkgo-project/ginkgo/pull/375)),
+ Fix SellP read function ([#374](https://github.com/ginkgo-project/ginkgo/pull/374)),
+ Add factorization support in `create_new_algorithm.sh`  ([#371](https://github.com/ginkgo-project/ginkgo/pull/371)).

### CODE REVIEW

File: core/device_hooks/cuda_hooks.cpp [hunk 1]
```diff
@@ -43,9 +43,9 @@ namespace gko {
 
 version version_info::get_cuda_version() noexcept
 {
-    // We just return 1.1.0 with a special "not compiled" tag in placeholder
+    // We just return 1.1.1 with a special "not compiled" tag in placeholder
     // modules.
-    return {1, 1, 0, "not compiled"};
+    return {1, 1, 1, "not compiled"};
```
Comment (upsj) [inline core/device_hooks/cuda_hooks.cpp]: Could we generate this from CMake, say using a `version.hpp.inc` and `configure_file`?

Comment (tcojean) [reply core/device_hooks/cuda_hooks.cpp]: This is done.

**Comment by tcojean** [comment]
@pratikvn I think adding that is rather complex as the changes are very tied to HIP and the CUDA code reorganization happened in between. In addition, as Mike says there will probably not be much change for now in overall performance, so I would rather leave this as is.

**Comment by pratikvn** [comment]
Okay. That makes sense.

File: cmake/Modules/FindPAPI.cmake [hunk 2]
```diff
@@ -67,7 +67,8 @@ if(PAPI_INCLUDE_DIR)
     endif()
 
     if (PAPI_LIBRARY)
-        # find the components
+       # find the components
+       enable_language(C)
        foreach(component IN LISTS PAPI_FIND_COMPONENTS)
            file(WRITE "${CMAKE_BINARY_DIR}/papi_${component}_detect.c"
```
Comment (yhmtsai) [inline cmake/Modules/FindPAPI.cmake]: line 70 - 73 needs one more space

**Comment by yhmtsai** [comment]
For documentation.
Ir solver is not shown in solver [link](https://ginkgo-project.github.io/ginkgo/doc/release/v1.1.1/group__solvers.html)
Factorizations does not show subitems in the graph [link](https://ginkgo-project.github.io/ginkgo/doc/release/v1.1.1/group__LinOp.html)

In preconditioner, it also shows gko::preconditioner::block_interleaved_storage_scheme used by jacobi and Preconditionable. Is it expected?

**Comment by tcojean** [comment]
@yhmtsai these should be fixed. There is also a `jacobi` documentation group btw, do I add a documentation header for this so that it appears as a module?

**Comment by tcojean** [comment]
@yhmtsai, can you check whether compilation still works on windows? Thanks

**Comment by yhmtsai** [comment]
@tcojean Sure, I can test it.
I need to wait my brother to open my laptop such that I can also test cuda kernel.

**Comment by pratikvn** [comment]
I just realized that we say in the tutorial that it is a 2D Poisson solver, whereas it is actually a 1D Poisson solver. We can just remove the 2D and just call it a Poisson solver. So, I think we should update the following:

1. The main [Doxygen page}(https://github.com/ginkgo-project/ginkgo/blob/develop/doc/DoxygenLayout.xml#L6)

2. The link in the [ Ginkgo webpage](https://ginkgo-project.github.io/) which redirects to the tutorial.

Any other places ?

**Comment by tcojean** [comment]
Reviews got dismissed again? Sorry about that, I guess I need to find a better way to generate documentation for the release branches.

**Pull request merged**